src/panoptes/pocs/focuser/birger.py

```diff
@@ -249,13 +249,15 @@ def _send_command(self, command, response_length=None):
         # Clear the input buffer in case there's anything left over in there.
         self._serial_port.reset_input_buffer()
 
-        # Send command
-        self._serial_io.write(command + '\r')
```

Comment:
You shouldn't take this out, otherwise you're trying to read the response before you've even sent the command once. This will cause a delay equal to the serial port timeout for no reason, every single time we send a command.

Reply:
Got it! Didn't realize that, I thought that reading without sending the command we would have gotten an `''`, so it'd get into the while loop and send the first command from there

Reply:
What I'm going to do is to move everything inside the loop, even the first time that the command is sent. That way I wouldn't be reading before sending the command for the first time.